### PR TITLE
Laboratory work #2, Alina Tryasoumova - 23FPL1

### PR DESCRIPTION
**Pull request by AlinaTR52**

### CODE REVIEW

**Comment by EIzharikov** [comment]
Unfortunately, there is nothing to watch. I advise you to hurry up and wish you good luck!

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -18,6 +18,25 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    word_list = text.split(' ')
+    for word_num in range(len(word_list)):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:24]: why so many for loops?? its just a tokenize function. I will accept this code only if this function will have 1 for loop.

File: lab_2_retrieval_w_bm25/main.py [hunk 2]
```diff
@@ -33,6 +52,22 @@ def remove_stopwords(tokens: list[str], stopwords: list[str]) -> list[str] | Non
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(tokens, list) or not isinstance(stopwords, list):
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:55]: join checks

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -33,6 +52,22 @@ def remove_stopwords(tokens: list[str], stopwords: list[str]) -> list[str] | Non
...
+    if not tokens or not stopwords:
+        return None
+    for word in tokens:
+        if not isinstance(word, str):
+            return None
+    for stopword in stopwords:
+        if not isinstance(stopword, str):
+            return None
+    for stopword in stopwords:
+        while stopword in tokens:
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:66]: ??

Comment (EIzharikov) [reply lab_2_retrieval_w_bm25/main.py:66]: i advice you to use list comprehension

**Review by EIzharikov — COMMENTED** [review]
A little progress. Rework tokenize and remove_stopwords functions and continue your work. Make CI green and be ready for next review. Good luck

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (AlinaTR52) [reply lab_2_retrieval_w_bm25/main.py]: I have rewritten this function

File: lab_2_retrieval_w_bm25/main.py (hunk 2 above)
Comment (AlinaTR52) [reply lab_2_retrieval_w_bm25/main.py]: fixed

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (AlinaTR52) [reply lab_2_retrieval_w_bm25/main.py:56]: fixed

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -18,6 +19,16 @@ def tokenize(text: str) -> list[str] | None:
...
     """
+    if not isinstance(text, str):
+        return None
+    letters_and_spaces = []
+    for letter in text:
+        if letter.isalpha() or letter == ' ':
+            letters_and_spaces.append(letter.lower())
+        else:
+            letters_and_spaces.append(' ')
+    tokenized_text = ''.join(letters_and_spaces).split()
```
Comment (EIzharikov) [inline lab_2_retrieval_w_bm25/main.py:30]: ok, but it can be better

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (EIzharikov) [reply lab_2_retrieval_w_bm25/main.py:57]: where?

**Review by EIzharikov — COMMENTED** [review]
So, you have unfixed comments. Make CI green, answer&fix all my comments and be ready for Friday. Good luck.

File: lab_2_retrieval_w_bm25/main.py (hunk 4 above)
Comment (AlinaTR52) [reply lab_2_retrieval_w_bm25/main.py]: I made it shorter

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (AlinaTR52) [reply lab_2_retrieval_w_bm25/main.py:51]: Sorry, I misunderstood your comment(

**Pull request closed without merging**